content/general/aws-authentication.textile

```diff
@@ -0,0 +1,212 @@
+</a>
+
+2. For type of trusted entity select "Another AWS account".
+
+<a href="/images/screenshots/iam/select-type-of-trusted-entity.png" target="_blank">
+  <img src="/images/screenshots/iam/select-type-of-trusted-entity.png" style="width: 100%" alt="Select type of trusted entity">
+</a>
+
+3. For Account ID specify 203461409171. This is the Ably AWS account.
+
+4. Click the "Require external ID checkbox" and then enter an external ID of @<Your_Ably_Account_ID>.<Your_Ably_app_ID>@. This is also displayed to you when you create an Ably Lambda Rule and select the "ARN of an assumable role" radio button in the create rule dialog.
+
+5. Click "Next: Permissions".
+
+6. You can create a Policy later, so click "Next: Tags".
+
+7. You don't need to add tags so click "Next: Review".
+
+8. Enter a suitable name for your role.
+
+<a href="/images/screenshots/iam/review-and-create-role.png" target="_blank">
+  <img src="/images/screenshots/iam/review-and-create-role.png" style="width: 100%" alt="Review and create Role">
+</a>
+
+9. Click "Create Role".
+
+h3(#set-trust-relationship). Set the trust relationship
+
+You have created a role. You now need to set the trust relationship for the role.
+
+1. Click "Roles" in the sidebar of the IAM console.
+
+2. Click on the role you just created.
+
+<a href="/images/screenshots/iam/select-your-role.png" target="_blank">
+  <img src="/images/screenshots/iam/select-your-role.png" style="width: 100%" alt="Select your Role">
+</a>
+
+3. Click the "Trust relationship" tab. 
+
+4. Click "Edit trust relationship".
+
+<a href="/images/screenshots/iam/trust-relationships.png" target="_blank">
+  <img src="/images/screenshots/iam/trust-relationships.png" style="width: 100%" alt="Trust relationships">
+</a>
+
+5. Check that the JSON is similar to as follows:
+
+```[json]
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::203461409171:root"
+      },
+      "Action": "sts:AssumeRole",
+      "Condition": {
+        "StringEquals": {
+          "sts:ExternalId": "<Your_Ably_Account_ID>.<Your_Ably_app_ID>"
+        }
+      }
+    }
+  ]
+}
+```
+
+Note, this should have been created for you, based on the information you provided when you created the role.
+
+6. Click "Cancel".
+
+You now have a role that will be used by the Ably AWS account, but your role will not have permissions to invoke your Lambda function. You need to create a suitable policy and attach it to this role.
+
+h3(#create-policy). Create a policy
+
+To create a policy:
+
+1. In the IAM console sidebar select "Policies"
+
+<a href="/images/screenshots/iam/create-policy.png" target="_blank">
+  <img src="/images/screenshots/iam/create-policy.png" style="width: 100%" alt="Create policy">
+</a>
+
+2. Click "Create Policy".
+
+3. Choose "Lambda" as the service. (Or other AWS service as applicable, such as AWS SQS, or AWS Kinesis).
```

Comment:
I think it's worth having explicit steps for each individual service (e.g. 3.1 for Lambda, 3.2 for SQS, 3.3 for Kinesis)

Reply:
Yes, I was wondering about that. The topic was already getting long, but I think now with a couple of sections removed this would work quite nicely. Thanks.